lib/Parse/ParseGeneric.cpp

```diff
@@ -368,6 +368,13 @@ ParserStatus Parser::parseGenericWhereClause(
     BodyContext.reset();
     HasNextReq = consumeIf(tok::comma);
     // If there's a comma, keep parsing the list.
+    // If there's a "&&", diagnose replace with a comma and keep parsing
+    if (Tok.isBinaryOperator() && Tok.getText() == "&&" && !HasNextReq) {
+      diagnose(Tok, diag::requires_comma)
+        .fixItReplace(SourceRange(Tok.getLoc()), ",");
```

Comment:
Nit: Not sure, but I think here you could just do `.fixItReplace(Tok.getLoc(), ",");` :)

Reply:
Really? I just imitated the code in `same-type-requirement` check:

```c
if (Tok.is(tok::equal)) {
  diagnose(Tok, diag::requires_single_equal)
     .fixItReplace(SourceRange(Tok.getLoc()), "==");
}
```
Though I really don't know what's the effect of `SourceRange`, If I remove `SourceRange` it seems still work fine.

Reply:
And then I find the definition of `fixItReplace`:

```c
InFlightDiagnostic &InFlightDiagnostic::fixItReplace(SourceRange R,
                                                     StringRef Str) {
  ...
}
```

So the first argument of `fixItReplace` seems need to be `SourceRange`. So I think `SourceRange(Tok.getLoc())` is necessary.

Reply:
You can see other examples of call like that if you search for `.fixItReplace(Tok.getLoc(),`
It ends up calling the same method, I think is just that `SourceRange` has a constructor that takes a `SourceLoc` the `SourceRange(Tok.getLoc())` is done implicitly :)

Reply:
You can just pass `Tok.getLoc()` which will construct a SourceRange using the loc as start and end of the range.

Reply:
yeah, the `SourceRange(Tok.getLoc())` call still exists, is just that the cpp compiler does it implicitly for us :)